src/compiler/checker.ts

```diff
@@ -5159,25 +5159,17 @@ namespace ts {
          * and if none of the base interfaces have a "this" type.
          */
         function isThislessInterface(symbol: Symbol): boolean {
-            for (const declaration of symbol.declarations) {
-                if (declaration.kind === SyntaxKind.InterfaceDeclaration) {
-                    if (declaration.flags & NodeFlags.ContainsThis) {
-                        return false;
-                    }
-                    const baseTypeNodes = getInterfaceBaseTypeNodes(<InterfaceDeclaration>declaration);
-                    if (baseTypeNodes) {
-                        for (const node of baseTypeNodes) {
-                            if (isEntityNameExpression(node.expression)) {
-                                const baseSymbol = resolveEntityName(node.expression, SymbolFlags.Type, /*ignoreErrors*/ true);
-                                if (!baseSymbol || !(baseSymbol.flags & SymbolFlags.Interface) || getDeclaredTypeOfClassOrInterface(baseSymbol).thisType) {
-                                    return false;
-                                }
-                            }
-                        }
-                    }
-                }
+            return !symbol.declarations.some(declaration =>
+                isInterfaceDeclaration(declaration) && (
+                    !!(declaration.flags & NodeFlags.ContainsThis)
+                    || some(getInterfaceBaseTypeNodes(declaration), n => !isThislessBaseType(n))));
+        }
+        function isThislessBaseType({ expression }: ExpressionWithTypeArguments): boolean {
+            if (!isEntityNameExpression(expression)) {
+                return true;

```

Comment:
I'm pretty sure this matches the logic from before, but it seems like this should be `false`?